### PR TITLE
update `hex_string` in parse_int fortran benchmark

### LINKED ISSUE #67: Improved Fortran version of hex_string
Suggested replacement for subroutine hex_string in file [perf.f90](https://github.com/JuliaLang/Microbenchmarks/blob/master/perf.f90) :

```fortran-free-form
! Convert an integer to a hex string
!
subroutine hex_string(dec,hexchar)
  integer, intent(in) :: dec
  character(*) :: hexchar

  integer :: i, quotient

  character(len=1), parameter :: table(0:15) = &
  [(char(i),i=ichar('0'),ichar('9')),(char(i),i=ichar('A'),ichar('F'))]

  quotient = dec
  hexchar  = '00000000'
  i = 8
  do while (quotient /= 0 .and. i > 0)
      hexchar(i:i) = table(iand(quotient,15))
      i = i-1
      quotient = ishft(quotient,-4)
  end do

end subroutine hex_string
```

On my PC, this replacement reduced the run time of the parse_integers benchmark from 0.130 s to 0.047 s.

### PR DESCRIPTION
**Pull request by oscardssmith**
changes based on https://github.com/JuliaLang/Microbenchmarks/issues/67

Closes #67

### CODE REVIEW

**Comment by oscardssmith** [comment]
@xecej4 any idea why this is failing the tests?

**Comment by acxz** [comment]
I believe this is failing due to the newer julia version. I need to redo some of the julia benchmarks with newer julia functionality.
Particularly trying to get rid of this compat file: https://github.com/JuliaLang/Microbenchmarks/blob/master/perfutil.jl

**Comment by xecej4** [comment]
Sorry, I have no idea what the testing procedure is or what tests are run. I am not familiar with Github (other than just downloading files from it). I did compile and run prog.f90 after making my suggested changes, using several Fortran compilers on my Windows PC and on a cloud Linux server, and the program ran normally, with less time used in the parse_string tests.


  *   Mecej4
  *

**Comment by acxz** [comment]
@oscardssmith sorry that it took me a while to get around to this, but if you could rebase this PR off the latest master, it should resolve the CI errors we are seeing.

@xecej4 thanks for the analysis on your local machine we should be able to confirm from the CI benchmarks as well. You can check the CI output yourself by clicking on the Details link. It will have a bunch of build/run output as well as the timings.

File: perf.f90 [hunk 1]
```diff
@@ -146,38 +146,22 @@ end function sysclock2ms
...
-
-  end do
+quotient = dec
+hexchar = '00000000'
+i = 8
+do while (quotient /= 0 .and. i > 0)
+hexchar(i:i) = table(iand(quotient,15))
+i = i-1
+quotient = ishft(quotient,-4)
+end do
```
Comment (acxz) [inline perf.f90:164]: Can we match the indentation and spacing as much as possible?

**Review by acxz — CHANGES_REQUESTED** [review]

**Comment by acxz** [comment]
CI passes and shows similar improvement as displayed by local results of @xecej4 

Given that the benchmark in question (parse_int) does not focus on how to write out the string in hex, this is a valid change to the benchmark and merging this in.

Thx for the contrib @xecej4 and for the help in pushing this along @oscardssmith !

**Pull request merged**